model/publishing/delivery/tasks/RemoteTaskStatusSynchroniser.php

```diff
@@ -28,33 +45,157 @@ public function getRemoteStatus()
 
     /**
      * @param $params
-     * @return \common_report_Report
+     * @return Report
+     * @throws \common_exception_Error
      * @throws \common_exception_MissingParameter
      */
     public function __invoke($params)
     {
-        if (count($params) != 2) {
+        if (count($params) != 4) {
             throw new \common_exception_MissingParameter();
         }
-        $taskId = array_shift($params);
-        $envId = array_shift($params);
+        list ($remoteTaskId, $envId, $deliveryUri, $testUri) = $params;
+        $this->remoteTaskId = $remoteTaskId;
+        $this->remoteEnvironmentId = $envId;
+
+        try {
+            $this->status = $this->fetchRemoteTaskStatus();
+            if (!$this->isTaskFinished($this->status)) {
+                return Report::createInfo(__('Remote task is still running.'));
```

Comment:
It's worth mentioning that the worker will re-queue the in case status is not finished. 
It's magic on https://github.com/oat-sa/tao-core/blob/master/models/classes/taskQueue/Worker/AbstractWorker.php#L124 
Perhaps is good to put this on Readme and also make a separate task for improve the way of re-queuing this tasks. 

It's scary we re-queue forever.